lib/id3c/utils.py

```diff
@@ -14,3 +16,46 @@ def wrap(function):
         function.__doc__ = function.__doc__.format_map(kwargs)
         return function
     return wrap
+
+
+def getattrpath(value: Any, attrpath: Union[str, Sequence[str]]) -> Any:
+    """
+    Get a nested named attribute, described by *attrpath*, from the
+    object *value*, or return None if one of the attributes in the path
+    doesn't exist.
+
+    *attrpath* may be a dotted-string like ``a.b.c`` or a sequence
+    (list or tuple) like ``("a", "b", "c")``.
+
+    >>> class Namespace:
+    ...     def __init__(self):
+    ...         self.__dict__ = {}
+    ...     def __repr__(self):
+    ...         return repr(self.__dict__)
+
+    >>> obj = Namespace()
+    >>> obj.a = Namespace()
+    >>> obj.a.b = Namespace()
+    >>> obj.a.b.c = 42
+    >>> getattrpath(obj, "a")
+    {'b': {'c': 42}}
+    >>> getattrpath(obj, "a.b.c")
+    42
+    >>> getattrpath(obj, "a.b.x")
+    >>> getattrpath(obj, "a.x")
+    >>> getattrpath(obj, "x")
+    >>> getattrpath(obj, ("a", "b", "c"))
+    42
+    """
+    if isinstance(attrpath, str):
+        attrpath = attrpath.split(".")
+
+    assert len(attrpath) != 0
+
+    for attr in attrpath:
+        value = getattr(value, attr, None)
+
+        if value is None:
+            return None
```

Comment:
My first version used recursion, but I then switched it to iteration. :-)

I considered and prototyped distinguishing between "present but None" and "not present", but we don't need to make that distinction currently so I punted on the additional complexity. Note that in this case at least one of `collectedDateTime` or `collectedPeriod` will be `None` since per-spec they're mutually exclusive.